content/Live Streaming Apps Architecture - Part 1 - German Lopez.md

```diff
@@ -0,0 +1,182 @@
+
+**Take Away No.3:** HLS and WebRTC are the options to have in mind when it comes to the playback with broad compatibility. It basically boils down to your latency needs.
+
+#### Scalability
+
+Making your platform/app able to support more concurrent users than the estimate is key. This is one of the success metrics for your live streaming app and means that your end users will not suffer from huge delays, video stuttering and connection drops when the numbers get high.
+
+Here is when the delivery formats play an important role. If we will be streaming to our end users via HLS(or any of the HTTP based protocols) we are one step closer to the goal. The fact that the media chunks are sent over HTTP makes these protocols way mor easy to scale. The solution recommended is to put a well-known Content Delivery Network (CDN) to do all the work for us.
+
+The high availability servers of the CDN will allow the delivery of our streaming to scale undefinitely.
+
+By doing this we are surpassing a couple of challenges at the same time. A CDN will reduce the number of connections to our live streaming servers, allowing us to make better use of those and get the costs down. Since CDNs have servers all around the globe, it will also reduce the latency for already cached chunks when requested from regions too far from the origin servers.
+
+**Most common cloud providers and their CDNs**
+- [AWS CloudFront](https://aws.amazon.com/cloudfront/)
+- [Akamai](https://www.akamai.com/us/en/products/media-delivery/)
+- [Azure CDN](https://azure.microsoft.com/en-us/services/cdn/#features)
+
+**Bonus:** [Wowza CDN](https://www.wowza.com/products/cdn) matches perfectly with the Wowza Streamin Engine software.
+
+<img src="images/CDN-arch.png" alt="Live Streaming Architecture using CDN with HLS"/>
+<br/>
+<br/>
+
+**Take Away No.4:** Delivering the content over HTTP is probably the best option for making your platform easily scalable.
+
+<br/>
+
+So, what happens if you choose to stream via RTMP, RTSP or WebRTC to the end users?
+Unfortunately, traditional CDNs aren't prepared for these protocols. We need to design our own server infrastructure to handle the load and scale properly, but don't worry; this is not a new problem.
+
+A broadly used concept is the **origin-edge** configuration for the core of the architecture. This defines a clear separation of responsibilities between the live streaming servers. A **origin** server would be the one in charge of ingesting the stream, and in some cases, also transcoding it to the delivery formats. On the other hand, **edge** servers will fetch the stream from the origin (or a designated transcoder) and distribute it to the end users.
+
+This allow us to use our computing resources more efficiently. In the case of 1:many streaming applications it also allows to distribute the connections between the edge servers, instead of overburden the origin.
```

Comment:
```suggestion
This allows us to use our computing resources more efficiently. In the case of 1 or more streaming applications, it allows for the distribution of connections between the edge servers, instead of overburdening the origin.
```

Reply:
1:many  should be like before or 1 to many since it is a relation indicator.